blocks/callout/callout.js

```diff
@@ -0,0 +1,69 @@
+export default async function decorate(block) {
+  const calloutData = {
+    title: block.children[0].children[0].textContent.trim(),
+    description: block.children[0].children[1].children,
+    url: block.children[1].children[0].textContent.trim(),
+    buttonLabel: block.children[1].children[1].textContent.trim(),
+    altText: block.children[1].children[2].textContent.trim(),
+  };
+
+  let calloutBlock;
+  let calloutButton = null;
+
+  // Create container element
+  const calloutContainer = document.createElement('div');
+  calloutContainer.classList = 'callout-container';
+
+  // Populate callout content
+  const calloutContent = document.createElement('div');
+  calloutContent.classList = 'callout__content';
+
+  // ...remove any automatic heading tag from title text
+  const calloutTitle = document.createElement('span');
+  calloutTitle.classList = 'callout__title';
+  calloutTitle.textContent = calloutData.title;
+  calloutContent.append(calloutTitle);
+
+  // ...remove semantic tags from description text
+  [...calloutData.description].forEach((textNode) => {
+    const calloutDescriptionPart = document.createElement('span');
+    calloutDescriptionPart.textContent = textNode.textContent.trim();
+    calloutContent.append(calloutDescriptionPart);
+  });
+
+  if (calloutData.url && calloutData.buttonLabel) {
+    // The callout is non-interactive, but has a link button
```

Comment:
what does this mean

Reply:
I updated the text for clarity! The comments for this block of code are now:

```js
// The callout has a button; the rest of the block is not interactive 
// The callout as a whole is interactive
// The callout is non-interactive
```